components/portal/widgets/controllers.js

```diff
@@ -457,53 +459,55 @@ define(['angular'], function(angular) {
     };
 
     var resolveTemplateStatus = function(now, actionStart, actionEnd) {
-      var result = '';
+      if (!now || !actionStart || !actionEnd) {
+        throw new TypeError('now, actionStart, and actionEnd must all be set');
```

Comment:
I'd rather not throw errors. Especially if they don't get caught.

If a widget is misconfigured it now would break the page, instead of just failing back to the basic widget view like it does in every other failure scenario.

Reply:
Semantically throwing a `TypeError` is correct here, however if it will break the app, it can be dropped to a blank `return`

Reply:
You're both right. The immediate function has constraints on its arguments. Passing it arguments that don't meet those constraints, that's an error. But if this function is going to error, anything that calls this needs to appropriately handle that error, so that errors in widgets don't escape that specific widget to wreck the whole page.